packages/design-system/src/atoms/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,25 @@
+import { useState, VFC } from 'react'
+
+import * as Styled from './Checkbox.styles'
+import { Props } from './Checkbox.types'
+
+export const Checkbox: VFC<Props> = ({ initiallyChecked = false, onCheck, color, label }) => {
+  const [isChecked, setChecked] = useState(initiallyChecked)
+
+  const handleCheck = (): void => {
+    setChecked((prevChecked) => !prevChecked)
+    onCheck(isChecked)
```

Comment:
Passing current state to callback function may cause UI and state desynchronization if a user spams mouse click on a checkbox. I would suggest calling the function inside the `setState` callback to avoid such situations.
```suggestion   
    setChecked((prevChecked) => {
    onCheck(isChecked)
    return !prevChecked
  })
```